kornia/augmentation/random.py

```diff
@@ -227,12 +227,12 @@ def random_affine_gen(
     else:
         shear_tmp = shear
 
-    transform: torch.Tensor = _get_random_affine_params(
-        batch_size, height, width, degrees_tmp, translate, scale, shear_tmp, same_on_batch)
-    return dict(transform=transform)
+    return _get_random_affine_params(
+        batch_size, height, width, degrees_tmp, translate, scale, shear_tmp, resample, same_on_batch)
 
 
 def random_rotation_gen(batch_size: int, degrees: FloatUnionType,
+                        interpolation: Resample = Resample.BILINEAR,
                         same_on_batch: bool = False) -> Dict[str, torch.Tensor]:
```

Comment:
To be honest, I do not think passing to GPU will make it faster since we are not doing intensive computations here. GPU is good at vector computations but please do remember that each core is pretty weak. I did a bit comparison here (torch GPU, torch CPU, numpy).
https://colab.research.google.com/drive/15okeDVpv_4H8dl8X6rzyi6BAcyNOPjbU

It shows that:
- Torch GPU: 400 us
- Torch CPU: 127 us
- NumPy: 30 us.

For the best param gen performance, I think NumPy is much better while it may break Kornia's philosophy.

Reply:
> I don't think interpolation should be in the signature since it does not affect in the generator

Yes. But it would have a better API consistency to put it in.



> Thinking about about maybe a more clear name might help like `random_rotation_generator`

This makes sense.